### PR TITLE
Added release 0.3 release notes and documentation

### PR DESCRIPTION
**Pull request by davecheney**
`release-0.3.md` will be removed before commiting this PR.

Signed-off-by: Dave Cheney <dave@cheney.net>

### CODE REVIEW

File: release-0.3.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# Contour 0.3
```
Comment (Bradamant3) [inline release-0.3.md]: Were you thinking to add this content to the readme? I assume that when you say you'll remove the file when you merge you don't mean you'll get rid of its contents ;). 
I'd recommend a new CHANGELOG.md file at the root; I'm happy to make the change when I copyedit if you like.

Comment (davecheney) [reply release-0.3.md]: I'll transpose this into the github release page, https://github.com/heptio/contour/releases

Comment (Bradamant3) [reply release-0.3.md]: got it, thanks

File: release-0.3.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+You can read more about [Contour's TLS support here][2].  
+
+### More supported annotations
+
+Support for the following annotations on the `service` and `ingress` object has been added in this release.
+
+- `kubernetes.io/ingress.allow-http: "false"` will remove the ingress configration from the non TLS listener. Envoy will not serve traffic for your ingress's vhost on port 80.
+- `ingress.kubernetes.io/force-ssl-redirect: "true"` will cause Envoy to issue an unconditional 301 redirect to the HTTPS version of your site.
+
+### gRPC API is not GA
```
Comment (Bradamant3) [inline release-0.3.md]: s/not/now --?

File: release-0.3.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+Support for Envoy's version 2 gRPC based API, introduced in Contour 0.2, has been marked GA.
+The previous REST API does not support new Envoy features such as SNI.
```
Comment (Bradamant3) [inline release-0.3.md]: rm "previous"

File: release-0.3.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+### More supported annotations
+
+Support for the following annotations on the `service` and `ingress` object has been added in this release.
+
+- `kubernetes.io/ingress.allow-http: "false"` will remove the ingress configration from the non TLS listener. Envoy will not serve traffic for your ingress's vhost on port 80.
+- `ingress.kubernetes.io/force-ssl-redirect: "true"` will cause Envoy to issue an unconditional 301 redirect to the HTTPS version of your site.
+
+### gRPC API is not GA
+
+Support for Envoy's version 2 gRPC based API, introduced in Contour 0.2, has been marked GA.
```
Comment (Bradamant3) [inline release-0.3.md]: s/has been/is

File: release-0.3.md [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+The previous REST API does not support new Envoy features such as SNI.
+The REST API is now deprecated and will be removed completely in Contour 0.4.
```
Comment (Bradamant3) [inline release-0.3.md]: Do you mean that Contour support for the REST API is deprecated? Or that the REST API is deprecated in Envoy? If the former (as I suspect), then edit:
"Support for the REST API is deprecated and will be removed in a future version of Contour."
(Unless we really can promise a thing for a specific version? I've been taught never to mention such things ...)

File: release-0.3.md [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+
+### Other improvements in this release
+
+- Contour no longer sends updates to Envoy periodically. Changes in the Kubernetes API are streamed to Envoy as they occur. In a steady state, no traffic flows from Contour to Envoy.
+- The address and port for Envoy's HTTP and HTTPS listeners are now configurable. This will be useful for anyone using a daemonset with host networking. This can also be used to force Envoy to bind to IPv6.
```
Comment (Bradamant3) [inline release-0.3.md]: s/daemonset/Daemonset

File: release-0.3.md [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+- Contour now supports the PROXY protocol to recover the remote IP of connections via an ELB in TCP mode. To enable this, add the `--use-proxy-protocol` flag to your `contour` deployment or daemonset's container's flags.
+- Update to client-go release 6.
+
+## Bug fixes (compared to Contour 0.2.1)
+
+- The `glog` library is now properly initalised on `contour serve` startup. Fixes #113. Thanks @willmadison 
+
+## Upgrading
+
+Contour 0.3 has made the YAML v2 bootstrap configuration format the default.
```
Comment (Bradamant3) [inline release-0.3.md]: s/has made/makes

File: release-0.3.md [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+In Contour 0.4 the JSON v1 bootstrap configuration option will be removed.
+Please consult the [upgrade notes][0] for how to update your deployment or daemonset manifests to the YAML bootstrap configuration format.
```
Comment (Bradamant3) [inline release-0.3.md]: rm "Please"
s/consult/Consult
s/deployment/Deployment
s/daemonset/Daemonset

File: release-0.3.md [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+
+Support for Envoy's version 2 gRPC based API, introduced in Contour 0.2, has been marked GA.
+The previous REST API does not support new Envoy features such as SNI.
+The REST API is now deprecated and will be removed completely in Contour 0.4.
+
+### Other improvements in this release
+
+- Contour no longer sends updates to Envoy periodically. Changes in the Kubernetes API are streamed to Envoy as they occur. In a steady state, no traffic flows from Contour to Envoy.
+- The address and port for Envoy's HTTP and HTTPS listeners are now configurable. This will be useful for anyone using a daemonset with host networking. This can also be used to force Envoy to bind to IPv6.
+- Contour now supports the PROXY protocol to recover the remote IP of connections via an ELB in TCP mode. To enable this, add the `--use-proxy-protocol` flag to your `contour` deployment or daemonset's container's flags.
```
Comment (Bradamant3) [inline release-0.3.md]: edit:
" ... add the `--use-proxy-protocol` flag to the flags for your `contour` Deployment or Daemonset.

File: release-0.3.md (hunk 5 above)
Comment (davecheney) [reply release-0.3.md]: It's definitely going away in 0.4

**Pull request merged**